### PR TITLE
Documentation Edit Pass

### PR DESCRIPTION
**Pull request by srvance**
I hope you don't mind. I noticed some typos as I was reading the docs to figure things out and decided to do a full edit pass once I was done.

This commit mostly addresses typos and minor punctuation issues with a couple larger rephrasings for clarity and one addition of detail.

### CODE REVIEW

File: addon/authenticators/base.js [hunk 1]
```diff
@@ -7,16 +7,17 @@ const { RSVP, Evented, Object: EmberObject } = Ember;
...
-  retrieving an access token, initiating authentication against an external
-  provider like Facebook etc. and depends on the specific authenticator. Any
-  data that the authenticator receives upon successful authentication and
-  resolves with from the
+  this might, e.g., post a set of credentials to a server and in exchange
+  retrieve an access token, initiating authentication against an external
+  provider like Facebook, etc. The details depend on the specific authenticator. Upon
+  successful authenticateion, any
+  data that the authenticator receives and
+  resolves via the promise returned from the
```
Comment (marcoow) [inline addon/authenticators/base.js]: This seems to be formatted a bit strange. In general, for all non-code text in the codebase we're adhering to a 80 chars limit per line.

File: addon/services/session.js [hunk 2]
```diff
@@ -18,6 +18,8 @@ const { computed, A, Service, Evented, getOwner, isNone, assert }  = Ember;
   });
   ```
 
+  and is __automatically injected by Ember Simple Auth's mixins__.
```
Comment (marcoow) [inline addon/services/session.js]: I wouldn't add this here as it might lead to confusion. In most apps, you'd only be including the mixins in 3 routes (`application`, sth. like `internal` and `login`) but would use the service in controllers etc. as well.

**Comment by marcoow** [comment]
> I hope you don't mind.

Certainly not ;)

File: addon/authenticators/base.js (hunk 1 above)
Comment (srvance) [inline addon/authenticators/base.js]: Fixed.

**Comment by srvance** [comment]
Rebased and resolved conflicts.

File: addon/mixins/application-route-mixin.js [hunk 3]
```diff
@@ -89,21 +89,29 @@ export default Mixin.create({
...
     const attemptedTransition = this.get('session.attemptedTransition');
+    const cookies = getOwner(this).lookup('service:cookies');
+    const redirectTarget = cookies.read('ember_simple_auth-redirectTarget');
 
     if (attemptedTransition) {
       attemptedTransition.retry();
       this.set('session.attemptedTransition', null);
+    } else if (redirectTarget) {
+      this.transitionTo(redirectTarget);
+      cookies.clear('ember_simple_auth-redirectTarget');
```
Comment (marcoow) [inline addon/mixins/application-route-mixin.js]: Hm, this is unrelated actually

File: addon/mixins/authenticated-route-mixin.js [hunk 4]
```diff
@@ -78,7 +81,15 @@ export default Mixin.create({
...
-      if (!this.get('_isFastBoot')) {
+      if (this.get('_isFastBoot')) {
+        const fastboot = getOwner(this).lookup('service:fastboot');
+        const cookies = getOwner(this).lookup('service:cookies');
+
+        cookies.write('ember_simple_auth-redirectTarget', transition.intent.url, {
+          path: '/',
+          secure: fastboot.get('request.protocol') === 'https'
+        });
+      } else {
```
Comment (marcoow) [inline addon/mixins/authenticated-route-mixin.js]: This is unrelated as well

Comment (srvance) [inline addon/mixins/authenticated-route-mixin.js]: Strange. I rebased against master and these didn't register as conflicts. Let me look into it.

Comment (srvance) [inline addon/mixins/authenticated-route-mixin.js]: Those came in as part of the rebase. They were changes on master after I had branched. There are no more changes for me to rebase. What am I missing? The failures seem to be related to a phantomjs install problem for Ember 1.12.

Comment (marcoow) [inline addon/mixins/authenticated-route-mixin.js]: Yeah, the test failures are unrelated to your changes - looks like travis had some problems for a few days now.

It seems like sth. went wrong when rebasing - your branch contains 2078be3336dcc64b7067d922233518e99e172de3 which has already been merged to master a few days ago…

Comment (srvance) [inline addon/mixins/authenticated-route-mixin.js]: Let's see how that goes. Some of the merges were non-trivial, which I think may have confused things (or at least me).

Comment (marcoow) [inline addon/mixins/authenticated-route-mixin.js]: Hm, there's still unrelated changes in here… Can you give me access to your fork? Then I can fix the rebase.

**Comment by srvance** [comment]
Done.

File: addon/mixins/authenticated-route-mixin.js (hunk 4 above)
Comment (srvance) [inline addon/mixins/authenticated-route-mixin.js]: You should have access.

**Comment by marcoow** [comment]
@srvance: rebased - can you check everything is correct still?

**Comment by srvance** [comment]
I think it looks good. If not, we can fix it! :)

**Comment by marcoow** [comment]
great - merging 🎉

**Comment by marcoow** [comment]
oops…
